test/unit/specs/bl/api-services.js

```diff
@@ -155,5 +155,66 @@ describe('<BusinessLogic> API Services', function() {
       })
 
     })
+
+    it('should run with options', async () => {
+      const req1 = prepareMockRequest()
+      const req2 = prepareMockRequest()
+      const req3 = prepareMockRequest()
+      const req4 = prepareMockRequest()
+
+      const httpRequestHeaders = { 'custom-header': 'headerValue' }
+      const options1 = { executionType: Backendless.BL.ExecutionTypes.ASYNC, httpRequestHeaders }
+      const options2 = { executionType: Backendless.BL.ExecutionTypes.ASYNC_LOW_PRIORITY, httpRequestHeaders }
+      const options3 = { executionType: Backendless.BL.ExecutionTypes.SYNC, httpRequestHeaders }
+
+      await Backendless.BL.CustomServices.invoke(serviceName, methodName, args, options1)
+
+      expect(req1).to.deep.include({
+        method : 'POST',
+        path   : `${APP_PATH}/services/${serviceName}/${methodName}`,
+        body   : args,
+        headers: {
+          ...httpRequestHeaders,
```

Comment:
in my opinion, would be better do not reuse any variables, because it can be modified inside the method and the test doesn't show us that. 

For Instance: 
1) the invoke method removes all the properties in the `httpRequestHeaders`
2) and in the line, the object will be empty
3) the actual request was sent without  the `custom-header` header, but the test is green

Reply:
Got you, made changes